scale/batch/messages/create_batch_recipes.py

```diff
@@ -91,6 +103,80 @@ def execute(self):
 
         self.new_messages.extend(new_messages)
         return True
+        
+    def _handle_new_batch(self, batch, definition):
+        """Handles creating a new batch of recipes with the defined dataset, returning any messages needed for the batch
+        
+        :param batch: The batch
+        :type batch: :class:`batch.models.Batch`
+        :param definition: The batch definition
+        :type definition: :class:`batch.definition.definition.BatchDefinition`
+        :return: The messages needed for the re-processing
+        :rtype: list
+        """
+        
+        messages = []
+
+        dataset = DataSet.objects.get(pk=definition.dataset)
+        dataset_definition = dataset.get_definition()
+        recipe_type = RecipeType.objects.get(pk=batch.recipe_type_id)
+        recipe_inputs = recipe_type.get_definition().get_input_keys()
+        
+        # No recipe inputs match the dataset 
+        if not any(elem in recipe_inputs for elem in dataset_definition.param_names):
+            return messages
+
+        # Get previous recipes for dataset files:
+        ds_files = DataSetFile.objects.get_dataset_files(dataset.id).values_list('scale_file_id', flat=True)
+        recipe_ids = RecipeInputFile.objects.filter(input_file_id__in=ds_files).values_list('recipe_id', flat=True)
```

Comment:
Not sure about reprocessing with a new batch. I think we only want to reprocess when superseding an old batch.  If we are reprocessing we need to also filter by recipe type and revision so we reprocess the correct recipe.